lib/fastlane/actions/mailgun.rb

```diff
@@ -0,0 +1,136 @@
+module Fastlane
+  module Actions
+    class MailgunAction < Action
+      def self.git_author
+        s = `git log --name-status HEAD^..HEAD`
+        s = s.match(/Author:.*<(.*)>/)[1]
+        return s if s.to_s.length > 0
+        return nil
+      rescue
+        return nil
+      end
+
+      def self.last_git_commit
+        s = `git log -1 --pretty=%B`.strip
+        return s if s.to_s.length > 0
+        nil
+      end
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      # As there is a text limit in the notifications, we are
+      # usually interested in the last part of the message
+      # e.g. for tests
+      def self.trim_message(message)
+        # We want the last 7000 characters, instead of the first 7000, as the error is at the bottom
+        start_index = [message.length - 7000, 0].max
+        message = message[start_index..-1]
+        message
+      end
+
+      def self.run(options)
+        require 'rest_client'
+
+        handle_exceptions(options)
+
+        options[:message] = self.trim_message(options[:message].to_s || '')
+
+        mailgunit(ENV['MAILGUN_APIKEY'],
```

Comment:
You have to access the configuration like this: `options[:mailgun_apikey]` to properly use both environment variables, parameters, default values and validation blocks.